bzt/jmx2yaml.py

```diff
@@ -0,0 +1,984 @@
+#! /usr/bin/env python
+"""
+Copyright 2015 BlazeMeter Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+"""
+import logging
+import os
+from bzt.six import etree
+import traceback
+import yaml
+from optparse import OptionParser
+from copy import deepcopy
+from collections import namedtuple
+from cssselect import GenericTranslator
+import sys
+from bzt.modules.jmeter import JMX
+
+KNOWN_TAGS = ["hashTree", "jmeterTestPlan", "TestPlan", "ResultCollector",
+              "HTTPSamplerProxy",
+              "ThreadGroup",
+              "kg.apc.jmeter.timers.VariableThroughputTimer",
+              "kg.apc.jmeter.threads.SteppingThreadGroup",
+              "DNSCacheManager",
+              "HeaderManager",
+              "CacheManager",
+              "CookieManager",
+              "ConfigTestElement",
+              "DurationAssertion",
+              "ConstantTimer",
+              "RegexExtractor",
+              "HtmlExtractor",
+              "com.atlantbh.jmeter.plugins.jsonutils.jsonpathextractor.JSONPathExtractor",
+              "com.atlantbh.jmeter.plugins.jsonutils.jsonpathassertion.JSONPathAssertion",
+              "ResponseAssertion",
+              "CSVDataSet",
+              "GenericController",
+              "ResultCollector"]
+
+
+class YML(JMX):
```

Comment:
class name does not reflect its responsibility. it's not YAML class! It's converter class... As we discussed, converter should produce dict. Some external to converter entity should dump it into YAML/JSON/whatever.